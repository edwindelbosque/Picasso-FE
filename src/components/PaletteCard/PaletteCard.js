import React from 'react';
import './PaletteCard.scss';
import Fade from 'react-reveal/Fade';

const PaletteCard = ({ menuIsActive, palette }) => {
	if (palette.id) {
		const colorForPalette = palette.colors.map((color, i) => {
			return <div key={i} style={{ background: color.hex.value }}></div>;
		});
		return (
			<Fade when={menuIsActive} duration={600} delay={150}>
				<div className='PaletteCard'>
					<h4>{palette.paletteName}</h4>
					<div className='colors'>{colorForPalette}</div>
				</div>
			</Fade>
		);
	} else {
		return (
			<> </>
		)
	}
};

export default PaletteCard;
