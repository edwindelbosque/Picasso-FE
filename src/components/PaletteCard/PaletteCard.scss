.PaletteCard {
	margin-top: 40px;
	cursor: pointer;
	transition: 0.1s ease-in-out;
	width: fit-content;
	background: rgba(0, 0, 0, 0);
	backdrop-filter: blur(10px);
	padding: 0 10px 10px 10px;
	border-radius: 4px;

	&:hover {
		box-shadow: 0 2px 7px rgba(0, 0, 0, 0.301);

		section {
			opacity: 1;
		}
	}

	section {
		border-radius: 50%;
		width: 30px;
		opacity: 0;
		height: 30px;
		background: rgb(228, 46, 46);
		color: white;
		position: absolute;
		z-index: 20;
		display: flex;
		align-items: center;
		justify-content: center;
		font-family: 'Roboto';
		font-weight: 300;
		transform: rotate(45deg);
		font-size: 25px;
		right: 10px;
		border: 0px solid black;
		top: 8px;
		transition: 0.2s;

		&:hover {
			background: rgb(255, 106, 106);
			transition: 0.1s;
		}
	}

	h4 {
		font-size: 30px;
		color: rgb(32, 42, 95);
		font-family: 'Roboto';
		padding: 6px 0;
		width: 100%;
		text-align: left;
	}

	.colors {
		display: flex;
		background: rgb(85, 85, 85);
		width: fit-content;
		justify-content: center;
		border-radius: 3px;
		box-shadow: 0 2px 5px rgba(0, 0, 0, 0.267);

		div {
			width: 125px;
			height: 125px;
		}
	}
}
