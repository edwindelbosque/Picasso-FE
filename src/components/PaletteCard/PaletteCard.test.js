import React from 'react';
import PaletteCard from './PaletteCard';
import { shallow } from 'enzyme';

describe('PaletteCard', () => {
	let wrapper;

	let palette = {
		id: 1,
		catalogName: 'Personal',
		user_id: 1,
		colors: [{ hex: 'green' }]
	};

	beforeEach(() => {
		wrapper = shallow(<PaletteCard palette={palette} />);
	});

	it('App should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
