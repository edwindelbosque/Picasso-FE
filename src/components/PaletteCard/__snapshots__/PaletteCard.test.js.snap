// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteCard App should match snapshot 1`] = `
<Fade
  delay={150}
  duration={600}
>
  <div
    className="PaletteCard"
  >
    <Link
      to="/catalogs/undefined"
    >
      <section
        onClick={[Function]}
      >
        +
      </section>
    </Link>
    <Link
      key="1"
      onClick={[Function]}
      to="/catalogs/undefined/palettes/1"
    >
      <h4 />
      <div
        className="colors"
      >
        <div
          key="12531"
          style={
            Object {
              "background": 12531,
            }
          }
        />
      </div>
    </Link>
  </div>
</Fade>
`;
